internal/manifests/collector/volume_test.go

```diff
@@ -76,3 +76,26 @@ func TestVolumeWithMoreConfigMaps(t *testing.T) {
 	assert.Equal(t, "configmap-configmap-test", volumes[1].Name)
 	assert.Equal(t, "configmap-configmap-test2", volumes[2].Name)
 }
+
+func TestVolumePrometheus(t *testing.T) {
```

Comment:
Can we add a test case where when prometheus is disabled we do not create the volumes

Reply:
This is done in the above test cases. When we don't provide the prometheus config, it only creates the volume for agent config.

Reply:
Can you make a dedicated unit-test, where check if prometheus is inside the volumes